source/common/http/http3/conn_pool.h

```diff
@@ -26,6 +37,41 @@ class ActiveClient : public MultiplexedActiveClientBase {
                                     data) {}
 };
 
+// Http3 subclass of FixedHttpConnPoolImpl which exists to store quic data.
+class Http3ConnPoolImpl : public FixedHttpConnPoolImpl {
+public:
+  Http3ConnPoolImpl(Upstream::HostConstSharedPtr host, Upstream::ResourcePriority priority,
+                    Event::Dispatcher& dispatcher,
+                    const Network::ConnectionSocket::OptionsSharedPtr& options,
+                    const Network::TransportSocketOptionsSharedPtr& transport_socket_options,
+                    Random::RandomGenerator& random_generator,
+                    Upstream::ClusterConnectivityState& state, CreateClientFn client_fn,
+                    CreateCodecFn codec_fn, std::vector<Http::Protocol> protocol,
+                    TimeSource& time_source)
+      : FixedHttpConnPoolImpl(host, priority, dispatcher, options, transport_socket_options,
+                              random_generator, state, client_fn, codec_fn, protocol) {
+    auto source_address = host_->cluster().sourceAddress();
+    if (!source_address.get()) {
+      auto host_address = host->address();
+      source_address = Network::Utility::getLocalAddress(host_address->ip()->version());
+    }
+    Network::TransportSocketFactory& transport_socket_factory = host->transportSocketFactory();
+    quic_info_ = std::make_unique<Quic::PersistentQuicInfoImpl>(
+        dispatcher, transport_socket_factory, time_source, source_address);
+    setQuicConfigFromClusterConfig(host_->cluster(), quic_info_->quic_config_);
+  }
+
+  Quic::PersistentQuicInfoImpl& quicInfo() { return *quic_info_; }
```

Comment:
Should this be a const ref, or do we intend callers to modify it?

Reply:
this is basically a container of things passed to quic client sessions as they're created.  since many of the items like the crypto config are non-const I think the into needs to be non-const as well.